### PR TITLE
IGNITE-19002 Move dataStorage configuration to distribution zones

### PR DESCRIPTION
**Pull request by kgusakov**
https://issues.apache.org/jira/browse/IGNITE-19002

### CODE REVIEW

File: modules/placement-driver/src/integrationTest/java/org/apache/ignite/internal/placementdriver/PlacementDriverManagerTest.java
Comment (ibessonov) [inline modules/placement-driver/src/integrationTest/java/org/apache/ignite/internal/placementdriver/PlacementDriverManagerTest.java:106]:
```suggestion
    @InjectConfiguration
```

File: modules/schema/src/test/java/org/apache/ignite/internal/schema/configuration/KnownDataStorageValidatorTest.java
Comment (ibessonov) [inline modules/schema/src/test/java/org/apache/ignite/internal/schema/configuration/KnownDataStorageValidatorTest.java:43]: Oh, so you didn't remove the unknown storage. Why?

File: modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/sql/DistributionZoneSqlDdlParserTest.java
Comment (ibessonov) [inline modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/sql/DistributionZoneSqlDdlParserTest.java:127]: Please add a description to the issue that you've created. It's empty now

File: modules/storage-page-memory/src/testFixtures/java/org/apache/ignite/internal/storage/pagememory/TestPersisteStorageConfigurationModule.java
Comment (ibessonov) [inline modules/storage-page-memory/src/testFixtures/java/org/apache/ignite/internal/storage/pagememory/TestPersisteStorageConfigurationModule.java:35]: Typo in the class name

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlZoneOptionEnum.java
Comment (korlov42) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlZoneOptionEnum.java:46]: Let's add this option to `org.apache.ignite.internal.sql.engine.sql.DistributionZoneSqlDdlParserTest#createZoneNoOptions`. Besides, since IgniteSqlZoneOption.key is now an abstract identifier, no need to maintain this enum anymore, because under `org.apache.ignite.internal.sql.engine.sql` we keep entities that is part of sql AST only.

It may be moved inside or close to DdlSqlToCommandConverter for convenience though

File: modules/sql-engine/src/main/codegen/includes/parserImpls.ftl
Comment (korlov42) [inline modules/sql-engine/src/main/codegen/includes/parserImpls.ftl:502]: also we need to remove all this keywords from config, because we don't need them anymore (keywords here are words enclosed in diamonds: PARTITIONS, REPLICAS, etc; see collections `keywords` and `nonReservedKeywords` in `modules/sql-engine/src/main/codegen/config.fmpp`)

File: modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneConfigurationParameters.java
Comment (kgusakov) [reply modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneConfigurationParameters.java:44]: This class is a simple container for delivering the zone configuration params from DDL to lower level. If any params is null, it will be replaced by defaults here org.apache.ignite.internal.distributionzones.DistributionZoneManager#createZone

File: examples/src/main/java/org/apache/ignite/example/storage/StorageEngineExample.java
Comment (kgusakov) [reply examples/src/main/java/org/apache/ignite/example/storage/StorageEngineExample.java:79]: In general, this keyword was introduced some months ago, not in this PR.

But current plan: to have the 2 different zones for primary and secondary storages. This topic is under discussion, though, I think

File: modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/configuration/DistributionZoneConfigurationSchema.java
Comment (kgusakov) [reply modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/configuration/DistributionZoneConfigurationSchema.java:60]: UnknownDataStorageConfigurationSchema is still there, also some obsolete (I think) machinery around it. But the scope of the current issue is a big enough already. Can I create the separate issue for removing the whole UnknownDataStorageConfigurationSchema from all tests and places? I can put it in the nearest sprint, I think, wdyt?

Comment (kgusakov) [reply modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/configuration/DistributionZoneConfigurationSchema.java:60]: But what about this validator - removed

File: modules/schema/src/test/java/org/apache/ignite/internal/schema/configuration/KnownDataStorageValidatorTest.java
Comment (kgusakov) [reply modules/schema/src/test/java/org/apache/ignite/internal/schema/configuration/KnownDataStorageValidatorTest.java]: The same https://github.com/apache/ignite-3/pull/1872#discussion_r1165974751

File: modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/sql/DistributionZoneSqlDdlParserTest.java
Comment (kgusakov) [reply modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/sql/DistributionZoneSqlDdlParserTest.java:128]: Done

File: modules/distribution-zones/src/testFixtures/java/org/apache/ignite/internal/distributionzones/DistributionZonesTestUtil.java
Comment (kgusakov) [reply modules/distribution-zones/src/testFixtures/java/org/apache/ignite/internal/distributionzones/DistributionZonesTestUtil.java]: done

File: modules/storage-api/build.gradle
Comment (kgusakov) [reply modules/storage-api/build.gradle]: After the moving to strage-api, removed this trick

File: modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/configuration/DistributionZoneConfigurationSchema.java
Comment (ibessonov) [reply modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/configuration/DistributionZoneConfigurationSchema.java]: If we stop using the "unknown" storage in current issue, I believe it would also be a right place to to remove everything related to it, like schemas, tests and validators.

File: examples/src/main/java/org/apache/ignite/example/storage/StorageEngineExample.java
Comment (ibessonov) [reply examples/src/main/java/org/apache/ignite/example/storage/StorageEngineExample.java:79]: What's a "secondary storage"? Never heard about it

File: modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java
Comment (vldpyatkov) [inline modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java:317]: It looks like a tricky way to do something.

File: modules/distribution-zones/src/testFixtures/java/org/apache/ignite/internal/distributionzones/DistributionZonesTestUtil.java
Comment (vldpyatkov) [inline modules/distribution-zones/src/testFixtures/java/org/apache/ignite/internal/distributionzones/DistributionZonesTestUtil.java:42]: Each parameter should start from a new line, in case of not enough characters for entire function definition in one line.

File: modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/configuration/DistributionZoneConfigurationSchema.java
Comment (kgusakov) [reply modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/configuration/DistributionZoneConfigurationSchema.java]: Done

File: modules/schema/src/test/java/org/apache/ignite/internal/schema/configuration/KnownDataStorageValidatorTest.java
Comment (kgusakov) [reply modules/schema/src/test/java/org/apache/ignite/internal/schema/configuration/KnownDataStorageValidatorTest.java]: Removed

File: modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java
Comment (ibessonov) [inline modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java:332]: You can probably just read the value from the configuration, but ok.
Why not `AtomicInteger` by the way?

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/ddl/DdlSqlToCommandConverter.java
Comment (ibessonov) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/ddl/DdlSqlToCommandConverter.java:520]: This could be a constant, why do we calculate it every time?

**Review by ibessonov — APPROVED** [review]
I think I'm good with these changes, thank you! Let's wait for the second approval then

**Review by korlov42 — APPROVED** [review]
I'm ok with the changes in `modules/sql-engine`

File: modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java
Comment (kgusakov) [reply modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java:332]: Just to additionally highlight, for reader, that concurrent access is not a target for this container.

**Pull request merged**
